### DISCUSSION TITLE
Internal services documentation

### DISCUSSION THREAD

**Discussion by kees-tce** [directus/directus#5088] [Archive]
Hi all,

Is there any documentation available on the internal services (exposed in custom hooks/endpoints)? Or any examples on how to update an item and or a file?

Thanks!

**Comment by rijkvanzanten** [comment]
Heya! We're about to set those up! Didn't want to document these internals too soon, as they're still being tweaked and finalized for use in extensions (to prevent exactly this type of issue 😁 : https://github.com/directus/directus/discussions/5060 ) (cc @nickrum)

**Reply by keesvanbemmel** [reply]
Thanks Rijk!

**Reply by arladmin** [reply]
Hi @rijkvanzanten. 

Could you please let me know if the docs (or even drafts) for the internals are available anywhere yet?
Want to get going with custom hooks and endpoints but having a hard trying to do anything related to the data in the db.

Should the JS SDK be used in the meanwhile or the internal services are the only/preferable method?

Thanks.

**Reply by rijkvanzanten** [reply]
> Should the JS SDK be used in the meanwhile or the internal services are the only/preferable method?

If you're in a hook/custom endpoint, using the services is the preferable method. The JS SDK will go over the network resulting in more overhead in the network traffic.

We don't have any reference on the internal services yet. In the meantime, the best reference is the `services` folder https://github.com/directus/directus/tree/main/api/src/services which contains all the services and their methods

**Reply by arladmin** [reply]
> In the meantime, the best reference is the services folder https://github.com/directus/directus/tree/main/api/src/services which contains all the services and their methods

@rijkvanzanten Thanks for this link! It's already quite helpful.
